components/renku_data_services/namespace/db.py

```diff
@@ -521,16 +569,6 @@ async def move_data_connector(
     ) -> None:
         """Rename or move a namespace."""
 
-        async def _check_ns_permissions(
-            user: base_models.APIUser, authz: Authz, ns: models.Namespace, scope: Scope
-        ) -> None:
-            """Helper function to check for namespace permissions."""
-            is_user_namespace = ns.kind == models.NamespaceKind.user
-            ns_id = ns.id if is_user_namespace else ns.underlying_resource_id
-            allowed = await authz.has_permission(user, ns.kind.to_resource_type(), ns_id, scope)
-            if not allowed:
-                raise errors.missing_or_unauthorized(ns.kind, ns.underlying_resource_id)
-
```

Comment:
this has been renamed as `_check_namespace_permissions`, minimally modified and moved to the top of the file.